crates/biome_html_analyze/src/lint/nursery/no_duplicate_classes.rs

```diff
@@ -0,0 +1,183 @@
+    /// This rule helps keep your class strings clean by detecting and auto-fixing duplicates.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```html,expect_diagnostic
+    /// <div class="flex flex"></div>
+    /// ```
+    ///
+    /// ```html,expect_diagnostic
+    /// <div class="p-4 text-red-500 p-4 bg-white"></div>
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```html
+    /// <div class="flex p-4"></div>
+    /// ```
+    ///
+    /// ```html
+    /// <div class="p-4 text-red-500 bg-white"></div>
+    /// ```
+    ///
+    pub NoDuplicateClasses {
+        version: "next",
+        name: "noDuplicateClasses",
+        language: "html",
+        sources: &[RuleSource::EslintBetterTailwindcss("no-duplicate-classes").same()],
+        recommended: false,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+/// State returned by the rule when duplicates are found.
+#[derive(Debug)]
+pub struct DuplicateClassesState {
+    /// The deduplicated class string.
+    pub deduplicated: Box<str>,
+    /// The list of duplicate class names found.
+    pub duplicates: Box<[Box<str>]>,
+    /// Whether the original string used single quotes.
+    pub is_single_quote: bool,
+}
+
+impl Rule for NoDuplicateClasses {
+    type Query = Ast<HtmlAttribute>;
+    type State = DuplicateClassesState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let attribute = ctx.query();
+
+        // Only check "class" attribute
+        let name = attribute.name().ok()?;
+        let name_token = name.value_token().ok()?;
+        if name_token.text_trimmed() != "class" {
+            return None;
+        }
+
+        // Get the attribute value
+        let initializer = attribute.initializer()?;
+        let value = initializer.value().ok()?;
+        let html_string = value.as_html_string()?;
+        let value_token = html_string.value_token().ok()?;
+        let value_text = value_token.text_trimmed();
+
+        // Check if single-quoted
+        let is_single_quote = value_text.starts_with('\'');
+
+        // Get the inner string (without quotes)
+        let inner_text = inner_string_text(&value_token);
+
+        // Split by whitespace and track duplicates
+        let mut seen: HashSet<&str> = HashSet::default();
+        let mut duplicate_set: HashSet<&str> = HashSet::default();
+        let mut deduplicated_parts: Vec<&str> = Vec::new();
+
+        for class in inner_text.text().split_whitespace() {
+            if seen.contains(class) {
+                duplicate_set.insert(class);
+            } else {
+                seen.insert(class);
+                deduplicated_parts.push(class);
+            }
+        }
+
+        if duplicate_set.is_empty() {
+            return None;
+        }
+
+        let deduplicated = deduplicated_parts.join(" ");
```

Comment:
This might be might not work. If the original string has newlines, this new string will contain only white spaces.